netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -8,7 +8,45 @@
 
 #define INITGUID
 
-#include "net_ebpf_ext.h"
+#include "net_ebpf_ext_xdp.h"
+
+//
+// Utility functions.
+//
+NTSTATUS
+_net_ebpf_extension_xdp_validate_if_index(uint32_t if_index)
+{
+    NTSTATUS status = STATUS_SUCCESS;
+    MIB_IPINTERFACE_ROW row = {0};
+    InitializeIpInterfaceEntry(&row);
+    row.InterfaceIndex = if_index;
+    // Try IPv4 first.
+    row.Family = AF_INET;
+    status = GetIpInterfaceEntry(&row);
+    if (!NT_SUCCESS(status)) {
+        // Try again with IPv6.
+        row.Family = AF_INET6;
+        status = GetIpInterfaceEntry(&row);
+    }
```

Comment:
This code looks wrong, xdp should still work even if neither IPv4 nor IPv6 is bound to the interface (at least when done at the NDIS layer instead of WFP).   At minimum, add a TODO comment that this should be fixed once XDP is fixed to support an attach type that doesn't require tcpip.

Reply:
Changed code to call `GetIfTable2` instead. This will work for interfaces not bound to Windows TCPIP stack. FYI, WFP L2 layer is implemented as a NDIS Lightweight filter. So it too will work for interfaces that are not bound to Windows TCPIP stack.